services/consensuscontext/validate_results_block_test.go

```diff
@@ -253,3 +265,38 @@ func MockProcessTransactionSetThatReturns(err error) func(ctx context.Context, i
 		return someEmptyTxSetThatWeReturnOnlyToPreventErrors, err
 	}
 }
+
+func TestConsensusContextValidateResultsBlock_TestBlockProposerNotSame(t *testing.T) {
```

Comment:
Test names are misleading - one is for Tx Block and the other for Rx Block, but that's not what the same say.

Reply:
names have been improved but the first test checks block-results-header against function input. Second checks block-results-header against block-tx-header.